overlord/servicestate/quota_handlers.go

```diff
@@ -823,7 +853,77 @@ func validateSnapForAddingToGroup(st *state.State, snaps []string, group string,
 			}
 		}
 	}
+	return nil
+}
+
+// splitSnapServiceName splits and verifies the snap service reference
+// taken in by the frontend. It expects the format snap.service
+func splitSnapServiceName(name string) (string, string, error) {
+	parts := strings.Split(name, ".")
+	if len(parts) != 2 {
+		return "", "", fmt.Errorf("invalid snap service: %s", name)
+	}
+	if err := naming.ValidateSnap(parts[0]); err != nil {
+		return "", "", err
+	}
+	return parts[0], parts[1], naming.ValidateApp(parts[1])
+}
+
+// ensureAppReferenceIsService returns whether the service referred to in the
+// snap is actually a service.
+func ensureAppReferenceIsService(st *state.State, snap, service string) error {
+	snapInfo, err := snapstate.CurrentInfo(st, snap)
+	if err != nil {
+		return err
+	}
+
+	appInfo, ok := snapInfo.Apps[service]
+	if !ok {
+		return fmt.Errorf("invalid service %q", service)
+	}
+	if !appInfo.IsService() {
+		return fmt.Errorf("%q is not a service", service)
+	}
+	return nil
+}
+
+// groupHasSubGroups returns true if the group has no sub-groups or it doesn't exist,
+// otherwise turns false.
+func groupHasSubGroups(group string, allGrps map[string]*quota.Group) bool {
+	grp, ok := allGrps[group]
+	if ok && len(grp.SubGroups) != 0 {
+		return false
+	}
+	return true
+}
```

Comment:
Since you are using this function from only one place, and it's short, I'd remove it. The main issue is that as a function it's a bit strange that it returns `true` if the group is invalid (though it's perfectly understandable if one looks where it's used).
Moving its contents into `validateSnapServicesForAddingToGroup` will make it more consequential.

Reply:
Not only that but the name was also inversed, thanks for the feedback!